### PR TITLE
Minor documentation improvements

### PR DESCRIPTION
**Pull request by lzutao**

### CODE REVIEW

File: src/elf/reloc.rs [hunk 1]
```diff
@@ -1,48 +1,70 @@
 //! # Relocation computations
+//!
+//! The following notation is used to describe relocation computations.
```
Comment (philipc) [inline src/elf/reloc.rs]: Note that these are specific to x86_64.

File: src/elf/reloc.rs [hunk 2]
```diff
@@ -1,48 +1,70 @@
...
+//!
+//!  * A: The addend used to compute the value of the relocatable field.
+//!  * B: The base address at which a shared object is loaded into memory
+//!       during execution. Generally, a shared object file is built with a
+//!       base virtual address of 0. However, the execution address of the
+//!       shared object is different. See Program Header.
```
Comment (philipc) [inline src/elf/reloc.rs]: Here and below: delete the 'See XXX' statements.

File: src/elf/compression_header.rs [hunk 3]
```diff
@@ -30,7 +30,7 @@ pub const ELFCOMPRESS_HIPROC: u32 = 0x7fff_ffff;
 macro_rules! elf_compression_header_std_impl { ($size:ty) => {
 
     #[cfg(test)]
-    mod test {
+    mod tests {
```
Comment (philipc) [inline src/elf/compression_header.rs]: What is the problem with the old name?

File: Cargo.toml [hunk 4]
```diff
@@ -14,15 +14,8 @@ include = ["src/**/*", "Cargo.toml", "CHANGELOG.md", "LICENSE", "README.md", "et
...
-[dependencies.log]
-version = "0.4"
-default-features = false
-optional = true
-
-[dependencies.scroll]
-version = "0.9"
-default_features = false
+log = { version = "0.4", default-features = false, optional = true }
+scroll = { version = "0.9", default-features = false }
```
Comment (philipc) [inline Cargo.toml]: Why? I'm not convinced that this is objectively better.

File: examples/dyldinfo.rs [hunk 5]
```diff
@@ -126,8 +122,7 @@ fn main () {
                     "-bind" => { bind = true }
                     "-lazy_bind" => { lazy_bind = true }
                     other => {
-                        println!("unknown flag: {}", other);
-                        println!("");
+                        println!("unknown flag: {}\n", other);
```
Comment (philipc) [inline examples/dyldinfo.rs]: Which clippy warning is this?

File: src/elf/compression_header.rs (hunk 3 above)
Comment (lzutao) [reply src/elf/compression_header.rs]: It's just a convention from [the Book](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-module-and-cfgtest).

File: Cargo.toml (hunk 4 above)
Comment (lzutao) [reply Cargo.toml]: I don't think one is better than another but the new one is shorter.
I'd revert if you prefer.

File: examples/dyldinfo.rs (hunk 5 above)
Comment (lzutao) [reply examples/dyldinfo.rs]: FYI, it is [print_with_newline](https://rust-lang.github.io/rust-clippy/master/index.html#print_with_newline).

File: src/elf/reloc.rs (hunk 2 above)
Comment (lzutao) [reply src/elf/reloc.rs]: Done

File: src/elf/reloc.rs (hunk 1 above)
Comment (lzutao) [reply src/elf/reloc.rs]: Done

File: Cargo.toml (hunk 4 above)
Comment (philipc) [reply Cargo.toml]: Yes, please drop that commit.

File: examples/dyldinfo.rs (hunk 5 above)
Comment (philipc) [reply examples/dyldinfo.rs]: Ah okay, please use `println!()` instead. It's clearer and I don't think the performance is a concern.

Comment (philipc) [reply examples/dyldinfo.rs]: Also I think you meant [println_empty_string](https://rust-lang.github.io/rust-clippy/master/index.html#println_empty_string).

Comment (lzutao) [reply examples/dyldinfo.rs]: If we revert this change, every time clippy run it will warn about this. 
Kind of annoyance.

File: Cargo.toml (hunk 4 above)
Comment (lzutao) [reply Cargo.toml]: Done.

File: examples/dyldinfo.rs (hunk 5 above)
Comment (lzutao) [reply examples/dyldinfo.rs]: Ok, I forget I could just drop the empty string.

**Pull request merged**